rmgpy/data/kinetics/rules.py

```diff
@@ -569,105 +583,144 @@ def __getAverageKinetics(self, kineticsList):
-                    # appended in the following lines.  Verbose comments are filtered out in 
-                    # rmgpy.rmg.model.CoreEdgeReactionModel.generateKinetics
-                    kinetics.comment = 'Average of [{0}]'.format(
-                         ' + '.join(k.comment if k.comment != '' else ';'.join(g.label for g in t) for k, t in kineticsList))
-                    kinetics.comment +='\n'
-                    # Append standard portion of kinetics comments that appear in non-verbose mode.
-                    kinetics.comment += 'Estimated using average of templates {0}'.format(
+                    kinetics.comment += 'Exact match found' 
+            else:
+                #Using a more general node to estimate original template
+                kinetics.comment +='Estimated using template ' + matchedLeaves
+                            
+        else:
+            # We found one or more results! Let's average them together
+            kinetics = self.__getAverageKinetics([k for k, t in kineticsList])
+            # Unlike in the case of a single rule, the verbose comments for averaging are lost unless they are 
+            # appended in the following lines.  Verbose comments are filtered out in 
+            # rmgpy.rmg.model.CoreEdgeReactionModel.generateKinetics
+            kinetics.comment = 'Average of [{0}]'.format(
+                    ' + '.join(k.comment if k.comment != '' else ';'.join(g.label for g in t) for k, t in kineticsList))
+            kinetics.comment +='\n'
+            # Append standard portion of kinetics comments that appear in non-verbose mode.
+            kinetics.comment += 'Estimated using average of templates {0}'.format(
                         ' + '.join([getTemplateLabel(t) for k, t in kineticsList]),
                     )
                 
-                kinetics.comment +=  ' for rate rule ' + originalLeaves
-                kinetics.A.value_si *= degeneracy
-                if degeneracy > 1:
-                    kinetics.comment += "\n"
-                    kinetics.comment += "Multiplied by reaction path degeneracy {0}".format(degeneracy)
+        kinetics.comment +=  ' for rate rule ' + originalLeaves
+        kinetics.A.value_si *= degeneracy
+        if degeneracy > 1:
+            kinetics.comment += "\n"
+            kinetics.comment += "Multiplied by reaction path degeneracy {0}".format(degeneracy)
+
+        return kinetics, entry if 'Exact' in kinetics.comment else None
 
-                return kinetics, entry if 'Exact' in kinetics.comment else None
+def removeIdenticalKinetics(kList):
+    """
+    doing this based on strings, which should be fine for this specifically, since we shouldn't have any
+    identical kinetics entries in the families all of the identical kinetics should look exactly the same
+    faster than the alternative
+    """
+    outSet = set()
+    outList = []
+    for i in xrange(len(kList)):
```

Comment:
As you never use `i` for anything except to look up the `k`, and you don't modify the `kList` inside the loop,  I suggest you could replace this with `for k in kList:`?